src/Compilers/CSharp/Portable/Symbols/PropertySymbolExtensions.cs

```diff
@@ -70,5 +70,16 @@ public static bool HasRefOrOutParameter(this PropertySymbol property)
             }
             return false;
         }
+
+        public static bool ShouldDisplayReadOnly(this IPropertySymbol property)
+        {
+            var getMethod = property.GetMethod as MethodSymbol;
+            var setMethod = property.SetMethod as MethodSymbol;
+
+            // If at least one accessor is present and all present accessors are readonly, the property should be marked readonly.
+            return (getMethod != null || setMethod != null) &&
```

Comment:
>getMethod != null [](start = 20, length = 17)

This should be `(object)... != null` or `... is object`, so that we don't invoke a `Symbol.Equals`.